pkg/middleware/auth.go

```diff
@@ -75,6 +75,13 @@ func RoleAuth(roles ...m.RoleType) macaron.Handler {
 
 func Auth(options *AuthOptions) macaron.Handler {
 	return func(c *m.ReqContext) {
+		if options.ReqSignedIn && c.License.HasLicense() && !c.License.HasValidLicense() {
```

Comment:
Would add this as a separate licensing middleware. Probably want to execute auth middleware before licensing as well.

Reply:
Yeah, I agree, this should probably be its own middleware.

Reply:
:+1: Is the approach of using middlewares for this (and adding the models.Licensing interface to the request context) reasonable in general?

Reply:
Think that's a great idea :+1: